sympy/polys/rootoftools.py

```diff
@@ -774,13 +774,58 @@ def _get_roots(cls, method, poly, radicals):
             if x.name not in free_names:
                 poly = poly.xreplace({d: x})
                 break
+
+        # compute rational lift if real algebraic
+        has_alg_coeff = any(c.is_algebraic and not c.is_rational
+                            for c in poly.all_coeffs())
+        has_alg_coeff = has_alg_coeff and all(c.is_real
+                            for c in poly.all_coeffs())
+
+        if has_alg_coeff:
+            poly_original = poly
+            poly = Poly(poly.expr, poly.gen, extension=True)
+            poly = poly.lift()
```

Comment:
There needs to be a way to prevent `extension = True` being used because it can be very slow. Sometimes it would be better to get an exception here rather than attempting to compute a primitive element.

The usual way of handling this would be to say that `real_roots` should accept `**kwargs` and then pass those through to the `Poly` constructor. Then a user would have to do `real_roots(..., extension=True)`. I wonder though whether for this method in particular it makes sense to have `extension = True` as the default though.